corehq/apps/locations/tests/test_reupholster.py

```diff
@@ -14,3 +17,70 @@ def test_path(self):
             parent = make_loc(name, type=type_, parent=parent)
         boston = parent
         self.assertEqual(boston.path, boston.sql_location.path)
+
+
+class TestNoCouchLocationTypes(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        LocationType.objects.create(domain='test-domain', name='test-type')
+
+    @classmethod
+    def tearDownClass(cls):
+        LocationType.objects.all().delete()
+
+    def setUp(self):
+        self.loc = Location(
+            domain='test-domain',
+            name='test-type',
+            location_type='test-type',
+        )
+        self.loc.save()
+
+    def tearDown(self):
+        delete_all_locations()
+
+    def test_change_location_type_name(self):
+        loc_type = LocationType.objects.create(domain='test-domain',
+                                               name='old-name')
+        loc = Location(
+            domain='test-domain',
+            name='Somewhere',
+            location_type='old-name'
+        )
+        loc.save()
+        loc_type.name = 'new-name'
+        loc_type.save()
+        # You need to look up the location from the db again, because the
+        # in-memory version stores the location_type it was created with
+        self.assertEqual(Location.get(loc._id).location_type, 'new-name')
+
+    def test_no_location_type(self):
+        with self.assertRaises(LocationType.DoesNotExist):
+            loc = Location(name="Something")
+            loc.save()
+
+    def test_type_set_correctly(self):
+        self.assertEqual(self.loc.location_type, 'test-type')
+        self.assertEqual(self.loc.sql_location.location_type.name, 'test-type')
+
+    def test_get_and_save(self):
+        # Get a location from the db, wrap it, access location_type, and save
+        loc = Location.get(self.loc._id)
+        self.assertEqual(loc.location_type, 'test-type')
+        loc.save()
+
+    def test_change_type_later(self):
+        new_type = LocationType.objects.create(domain='test-domain',
+                                               name='new-type')
+        self.loc.location_type = 'new-type'
+        self.loc.save()
+        self.assertEqual(self.loc.location_type, 'new-type')
```

Comment:
small but i think this test should get the object from the DB for the same reason described here: https://github.com/dimagi/commcare-hq/pull/7237/files#diff-53a1056b082a6e03761ea032f213dee1R54